File: 04-estructuras-datos/01-fundamentos-y-complejidad/scala/practicas/CompararOperaciones.scala
object CompararOperaciones:
  def main(args: Array[String]): Unit =
    val datos = Vector(10, 20, 30, 40)
    // El índice accede a una posición concreta de Vector.
    println(datos(2))
    // contains expresa una búsqueda por valor dentro de la colección.
    println(datos.contains(40))

File: 04-estructuras-datos/01-fundamentos-y-complejidad/java/practicas/CompararOperaciones.java
import java.util.ArrayList;
import java.util.List;

public class CompararOperaciones {
    public static void main(String[] args) {
        List<Integer> datos = new ArrayList<>(List.of(10, 20, 30, 40));
        // Acceder por índice no exige recorrer explícitamente los elementos anteriores.
        System.out.println(datos.get(2));
        // Buscar por valor puede revisar varios elementos hasta encontrar una coincidencia.
        System.out.println(datos.contains(40));
        // Insertar al inicio desplaza los elementos existentes de ArrayList.
        datos.add(0, 5);
        System.out.println(datos);
    }
}
